bin/hdfcoinc/pycbc_coinc_statmap

```diff
@@ -146,7 +219,160 @@ else:
     f['foreground/ifar_exc'] = numpy.array([])
     f['foreground/fap_exc'] = numpy.array([])
 
-if 'name' in d.attrs:
-    f.attrs['name'] = d.attrs['name']
+if 'name' in all_trigs.attrs:
+    f.attrs['name'] = all_trigs.attrs['name']
+
+# Incorporate hierarchical removal for any other loud triggers
+logging.info("Instantiating hierarchical removal.")
+
+# Step 1: Check to see if the trigger with the largest FAR is
+#         louder than all of the background.
+# fnlouder is the number of background events that are louder than
+# the foreground trigger. So if a trigger is louder than all of
+# the background, fnlouder = 0.
+
+loud_t_ind = []
+count_ind = -1
+for i in range(0, len(fnlouder)):
+    if fnlouder[i] == 0:
+        loud_t_ind.append(i)
+        count_ind += 1
+
+# Step 2 : Loop until we don't have to hierarchically remove
+#          anymore.
+iter = 1
+while len(loud_t_ind) > 0 and iter <= args.max_hier_removal :
+    # find the loudest of these triggers, its index, its newsnr
+    # and its ifar
+    max_ifar = 0
+    max_ifar_ind = 0
+    max_rank_stat = 0
+    for i in range(0, len(loud_t_ind)):
+         # Check for the largest ifar
+         if ifar[loud_t_ind[i]] >= max_ifar:
+           # Check to see if they have equal ifar, then check largest
+           # ranking statistic
+              if ifar[loud_t_ind[i]] == max_ifar:
+                   if back_stat[loud_t_ind[i]] >= \
+                      back_stat[loud_t_ind[max_ifar_ind]] :
+                        max_ifar_ind = loud_t_ind[i]
+                        max_rank_stat = back_stat[loud_t_ind[i]]
+                   else :
+                        max_rank_stat = back_stat[max_ifar_ind]
+                        
+              else :
+                   max_ifar = ifar[loud_t_ind[i]]
+                   max_ifar_ind = loud_t_ind[i]
+    # Step 3: Remove that trigger from the list of zerolag triggers
+
+    # FIXME:If the trigger is really the loudest trigger in the bin, then it's
+    # ranking statistic is the largest and also unique (I think).
+    where_is_fore_t = np.where(all_trigs.stat[:] == fore_stat[max_ifar_ind])
+
+    if len(where_is_fore_t[0]) > 1:
+         print "The author of this hierarchical removal is not a good ", \
+               "programmer. He is looking for a mapping from loudest ", \
+               "foreground trigger to the array of all triggers and ", \
+               "he assumes that there's only one matching SNR trigger ", \
+               "so this code's going to break."
+
```

Comment:
I'm pretty concerned about this section, but I don't think it will fail. I try to find the loudest foreground trigger that is louder than the background, but making the mapping from the list of foreground trigger to the list of all triggers (foreground and background) is done using this numpy.where. That has the potential of grabbing another trigger other than the one you want to remove. I can't think of a situation where it would grab a trigger that isn't the loudest one since if it's louder than all the background and louder than all the other foreground triggers, you can't grab another trigger with the same ranking statistic.

But people should be made aware of this.